### PR TITLE
Implement the Thin ICE design in CHURN.

### PR DESCRIPTION
**Pull request by bemasc**

### CODE REVIEW

File: src/churn/churn.ts [hunk 1]
```diff
@@ -55,12 +53,77 @@ module Churn {
...
+  }
+
+  export var selectPublicAddress =
+      (candidates:freedom_RTCPeerConnection.RTCIceCandidate[])
+      : NatPair => {
+    var address :string;
+    var port :number;
+    for (var i = 0; i < candidates.length; ++i) {
+      var line = candidates[i].candidate.trim();
+      var tokens = line.split(' ');
```
Comment (trevj) [inline src/churn/churn.ts]: It seems like you could use `splitHostCandidateLine_` to split the line and ensure it's a candidate line (and check here that it's a UDP candidate).

File: src/churn/churn.ts [hunk 2]
```diff
@@ -117,25 +195,38 @@ module Churn {
...
+        });
+      });
+
+      // Start the obfuscated connection.
+      this.configureObfuscatedConnection_(config);
+
+      // Once the obfuscated connection's local endpoint is known, the remote
+      // peer has sent us its public endpoint, and probing is complete, we can
+      // configure the obfuscating pipe and allow traffic to flow.
       // the information we need in order to configure the pipes required to
```
Comment (trevj) [inline src/churn/churn.ts]: think you can remove this and the next line

**Comment by trevj** [comment]
Very nice. :+1: once you get a chance to address the couple of comments I had.

File: src/churn/churn.ts (hunk 2 above)
Comment (bemasc) [inline src/churn/churn.ts]: Done.

File: src/churn/churn.ts (hunk 1 above)
Comment (bemasc) [inline src/churn/churn.ts]: Done.

**Pull request merged**